docs/development.md

```diff
@@ -1,54 +1,113 @@
 # Oak Development
 
+- [Meta-Advice](#meta-advice)
+- [Prerequisites](#prerequisites)
+- [Run Example Application](#run-example-application)
+  - [Build Application](#build-application)
+  - [Build Runtime Server](#build-runtime-server)
+  - [Run Runtime Server](#run-runtime-server)
+  - [Run Example Client](#run-example-client)
+
+## Meta-Advice
+
+For any open-source project, the best way of figuring out what prerequisites and
+dependendencies are required is to look at the projects continuous integration
+(CI) scripts. A project's top-level [Travis](https://travis-ci.org/),
+[CircleCI](https://circleci.com/),
+[Cloudbuild](https://cloud.google.com/cloud-build) or
+[Dockerfile](https://docs.docker.com/) will give definitive information about
+what's needed, and at what version.
+
+For Oak, the CI system is currently Cloudbuild using a Docker image, so the
+top-level [`cloudbuild.yaml`](/cloudbuild.yaml) and [`Dockerfile`](Dockerfile)
+hold the project's configuration, dependencies and build scripts.
+
 ## Prerequisites
 
-We use Docker to install the base dependencies that need to exist at the system
-level, e.g. the [Rust compiler](https://www.rust-lang.org/tools/install) and the
-[Bazel build system](https://bazel.build); these steps are detailed in
-[`Dockerfile`](/Dockerfile). See
-https://docs.docker.com/engine/reference/builder/ for more information.
-
-Inside Docker, we use Bazel to version, install and build any non-Rust
-dependencies, including our own code. Dependencies are listed in
-[`WORKSPACE`](/WORKSPACE). See
-https://docs.bazel.build/versions/master/external.html for more information.
-
-To set up your development environment, you need the following applications. For
-the accurate versions required for a successful build please consult the current
-[`Dockerfile`](/Dockerfile).
-
-- **Docker**: follow
-  [Docker install instructions](https://docs.docker.com/install)
-- **Bazel**: follow
-  [Bazel install instructions](https://docs.bazel.build/versions/master/install.html)
-- **Rust**:
-  - Follow install instructions from https://rustup.rs/:
+The project's [`Dockerfile`](/Dockerfile) holds the definitive set of
+prerequisites for building and running Oak. The steps involved are generally
+clear from the contents, but the
+[Docker documentation](https://docs.docker.com/engine/reference/builder/) gives
+details for any steps that aren't obvious.
```

Comment:
I think those two paragraphs could be merged, since they seem to describe the same thing. 

On kind of a higher level perspective, I worked with projects with more extensive documentation before, and I generally think it's a good idea. 

I completely understand that maintaining human language docs can often feel like maintaining a [prose copy of code](https://github.com/project-oak/oak/pull/931#pullrequestreview-404918403), but it does lower the bar for new contributors. 

That said, code-as-documentation does reduce developer overhead, and seems to be the preferred Google approach. Either way is fine, but good to be conscious of. :)

Reply:
That makes sense. :) 

I think basically a choice between speed of development/reviewing for us (ideally whenever code is changed, reviewers point to updating docs, which can be tedious), and ease of getting started for new contributors. It's absolutely fine to choose the former right now. 

As the project matures more, and we become more interested in plenty of less familiar external people using Oak, the docs will probs raise in importance.

Reply:
The reason I like to mention this, is that I think there's a difference between what works for Google internally, and what's important for a (mature) OSS project. 

Internally Google seems to rely mostly on code-as-docs. This makes a lot of sense, since we all go through a lenghtly noogler period to ramp up. It's fair to expect that we know of to navigate the Google codebase to learn how to use things. 

Externally of course, with (mature) OSS projects that people may spend a very brief time evaluating/testing things look different.